hw/chisel_acc/src/main/scala/snax_acc/reshuffle/MaxPoolingPE.scala

```diff
@@ -0,0 +1,64 @@
+package snax_acc.reshuffle
+
+import chisel3._
+import chisel3.util._
+
+// control signals for the basic processing element, related to the pooling algorithm
+class MAXPoolPECtrlIO(params: ReshufflerParams) extends Bundle {
+
+  val init_i = Input(Bool())
+
+}
+
+// data signals for the basic processing element, related to the pooling algorithm
+class MAXPoolPEDataIO(params: ReshufflerParams) extends Bundle {
+
+  val input_i = Flipped(Decoupled(SInt(params.inputWidth.W)))
+  val output_o = Decoupled(SInt(params.outputWidth.W))
+
+}
+
+// processing element input and output declaration
+class MAXPoolPEIO(params: ReshufflerParams) extends Bundle {
+
+  val ctrl = new MAXPoolPECtrlIO(params)
+  val data = new MAXPoolPEDataIO(params)
+
+}
+
+// MAXPoolPE module
+class MAXPoolPE(params: ReshufflerParams)
+    extends Module
+    with RequireAsyncReset {
+  val io = IO(new MAXPoolPEIO(params))
+
+  val resReg = RegInit(0.S(params.inputWidth.W))
+  // the receiver isn't ready, needs to send several cycles
+  val keep_output = RegInit(0.B)
+
+  when(io.data.input_i.valid) {
+    when(io.ctrl.init_i) {
+      // max pooling
+      resReg := Mux(io.data.input_i.bits > -128.S, io.data.input_i.bits, -128.S)
```

Comment:
Why is this here? Like what is the `init_i` for? 

So when `init_i` it returns either the input or the lowest value?